pkg/consensus/consensus.go

```diff
@@ -510,3 +511,23 @@ func convertKeys(validators []crypto.PublicKey) (pubs []*keys.PublicKey) {
 
 	return
 }
+
+func newBlockFromContext(ctx *dbft.Context) block.Block {
+	block := new(neoBlock)
+	if ctx.TransactionHashes == nil {
+		return nil
+	}
+
+	block.Block.Timestamp = uint32(ctx.Timestamp / 1000000000)
+	block.Block.Index = ctx.BlockIndex
+	block.Block.NextConsensus = ctx.NextConsensus
+	block.Block.PrevHash = ctx.PrevHash
+	block.Block.Version = ctx.Version
+	block.Block.ConsensusData = ctx.Nonce
+
+	if len(ctx.TransactionHashes) != 0 {
+		mt := merkle.NewMerkleTree(ctx.TransactionHashes...)
+		block.Block.MerkleRoot = mt.Root().Hash
+	}
+	return block
```

Comment:
Just a question, is the block without transactions valid here and don't we need to fill MerkleRoot in any case?

Reply:
So, perhaps, we should add one more check (L517-519)?
```
if ctx.TransactionHashes == nil || len(ctx.TransactionHashes) == 0 {
		return nil
}
```

Reply:
I guess the second check also covers the first one, but you're right, it'd be appropriate here, although this code was essentially just a copy-paste from the one that was in the dbft package before 3.0 changes.